STAC-24598: Add release notes 2.10.0 release and update with staging changes (#287)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/release-notes/v2.10.0.adoc

= v2.10.0 - 12/May/2026
:page-languages: [en, de, es, fr, ja, pt, zh]
:revdate: 2026-05-12
:page-revdate: {revdate}
:description: SUSE Observability Self-hosted

== Release Notes: {stackstate-product-name} Helm Chart v2.10.0

== Component Versions

[cols="1,2"]
|===
| Component | Version

| *SUSE Observability* | `2.10.0`
| *SUSE Observability Agent* | `1.3.18`
|===

== Upgrade Warnings & Breaking Changes

* *Helm Chart Structure Refactoring:* The Helm chart `values.yaml` has undergone a major refactoring to improve readability and maintainability. 

> ** The `templates/` directory is now reorganized into per-component subfolders.

Reviewer: is this really a user-facing important change?

> ** PVC, RBAC, and s3proxy resources now utilize the `common` library chart.

Reviewer: same here, not sure how actionable thjis is to an end-user

** ClickHouse backup resources (CronJob, Service, ConfigMaps) have moved into the `clickhouse` subchart. 
** The `*.enabled` toggles for ClickHouse, HBase, and OpenTelemetry have been removed; these subcharts now install unconditionally.
** *Action Required:* Before upgrading, you must review the xref:setup/release-notes/v2.10.0-helm.adoc[Helm chart migration guide for v2.10.0] and update your `values.yaml` accordingly.
* *Victoria Metrics Storage Format Change:* Victoria Metrics `v1.133.0` introduces a *non-reversible* storage format change (per-partition index). 
** On the first startup after the upgrade, Victoria Metrics will re-index all time series. You should expect slower ingestion and query performance during this one-time migration. 
** *Action Required:* A full backup *before* upgrading is mandatory. Rolling back to `v1.109.0` is impossible without restoring from a pre-upgrade backup.
* *Minio Replaced by S3Proxy:* Minio has been replaced with S3Proxy as the storage proxy for backups. 

> ** *Action Required:* Follow the xref:setup/data-management/backup_restore/migration_to_s3proxy.adoc[migration to S3Proxy guide] to update your configuration. This is a breaking change for IRSA users (the service account name has changed). For other setups, the existing configuration is deprecated but still supported. You must also download the latest version of the `sts-backup` CLI tool when restoring backups.

Reviewer: we need to sift through this to see what is relevant for the cusrtomer and whether it needs to be the first thing a user reads.

** *VEX Transparency for Actionable Scans:* To assist your security teams in managing vulnerability data, we are now providing VEX (Vulnerability Exploitability eXchange) data. This allows your scanners to identify which vulnerabilities are actually exploitable, ensuring your reports stay focused on real, actionable risks rather than "noise."
** *Optimized Dependency Lifecycle:* We have refined our build pipeline to ensure that the libraries and dependencies within our images are updated with the latest security patches at the time of release.
** *Continuous Security Pipeline:* These improvements are now a permanent part of our CI/CD process, ensuring that both our "Daily" and "Stable" releases maintain an ultra-low CVE profile as a continuous standard.

== New Features & Enhancements

* *StackPack Version Management:* Users can now manually remove old, uploaded StackPack versions. Additionally, the system will automatically clean up old StackPack versions that are no longer in use.
* *VictoriaMetrics Smart Backups:* Replaced the single-generation incremental backup system with a more robust two-tier smart backup strategy based on the VictoriaMetrics smart backups pattern.

> * *HA Profile Scaling Updates:* ** *500-ha Profile:* Updated the memory settings for the HBase region server and Elasticsearch.

Reviewer: this needs to be on a new line

** *4000-ha Profile:* Now deploys 5 datanodes with increased replication to improve availability and resilience.

> * *Overview Page Performance:* Improved the loading performance of the Platform overview page.

Reviewer: this needs to be the headline feature

* *RBAC Clustered Scope Metrics:* Fixed an issue where cluster-level metrics were hidden from RBAC users with a clustered scope.
* *Duplicate Monitors:* Fixed an issue with duplicate Node Ready, CPU, and Disk monitors. *(Note: This fix will temporarily show a duplication immediately upon upgrade, after which it will resolve.)*
* *User Login via External IDP:* Fixed a `Too many elements (XXX > 256) specified in within clause` error that occurred when logging in users belonging to more than 256 groups. Ensure the login page correctly displays after logout when using an external identity provider on non-SaaS instances.
* *System Load Metrics:* Corrected the metric used to calculate the 15-minute System Load on nodes.
* *RBAC Agent Resources:* Increased memory resources and set default CPU resources for the RBAC agent.

=== UI & Navigation
* *Y-Axis Ranges:* Ensured that Y-axis minimum and maximum range settings are now consistently applied in charts.
* *Browser Navigation:* Navigation history is now correctly maintained per individual browser tab.
* *Monitor Sort Order:* Established a stable, predictable sort order for monitors on the component highlight page.

=== Backup & Restore

> * *StackPack Backups:* Backups now correctly include all active StackPacks, including manually uploaded ones. Previously, only the configuration was backed up, forcing users to re-upload the StackPacks themselves after a restore.

Reviewer 2: ```suggestion
* *StackPack Backups:* Backups now correctly include all active StackPacks, including manually uploaded ones. Previously, only the configuration was backed up, requiring a manual upload of the StackPacks after a restore.
```
